### PR TITLE
mgr/dashboard: Device health status is not getting listed under hosts section

### PR DESCRIPTION
**Pull request by aaSharma14**
Device health is shown as failed to retrieve data under Hosts > Device Health section. This PR intends to fix this issue.

Fixes: https://tracker.ceph.com/issues/49354
Signed-off-by: Aashish Sharma <aasharma@redhat.com>

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by aaSharma14** [comment]
jenkins test dashboard

**Review by alfonsomthd — CHANGES_REQUESTED** [review]
Can you provide test coverage?

**Comment by aaSharma14** [comment]
@Angadi

> Can you provide test coverage?

Modified existing test for the new change.

File: src/pybind/mgr/dashboard/tests/test_ceph_service.py [hunk 1]
```diff
@@ -85,14 +85,16 @@ def _get_smart_data(d):
...
+        ('daemon', ('osd', '1'), 'with ID 1'),
+        ('daemon', ('mon', '2'), 'with ID 2')
     ]
 )
 def test_get_smart_data(caplog, by, args, log):
     # pylint: disable=protected-access
     expected_data = {
         'aaa': {'device': {'name': '/dev/sda'}},
         'bbb': {'device': {'name': '/dev/sdb'}},
+        'ccc': {'device': {'name': '/dev/sdc'}},
```
Comment (alfonsomthd) [inline src/pybind/mgr/dashboard/tests/test_ceph_service.py:97]: This is not testing your fix as the fixed method is already patched.
A test is needed to check that the appropriate ceph command is called depending on device type.
Something like this:
```
diff --git a/src/pybind/mgr/dashboard/tests/test_ceph_service.py b/src/pybind/mgr/dashboard/tests/test_ceph_service.py
index 6ff2199ade9..5b5fe5e1cf5 100644
--- a/src/pybind/mgr/dashboard/tests/test_ceph_service.py
+++ b/src/pybind/mgr/dashboard/tests/test_ceph_service.py
@@ -109,3 +109,22 @@ def test_get_smart_data(caplog, by, args, log):
             CephService._get_smart_data_by_device.assert_not_called()
             assert smart_data == {}
             assert log in caplog.text
+
+
+@mock.patch.object(CephService, 'send_command')
+def test_get_smart_data_from_appropriate_ceph_command(send_command):
+    send_command.side_effect = [
+        {'nodes': [{'name': 'osd.1', 'status': 'up'},{'name': 'mon.1', 'status': 'down'}]},
+        {'fake': {'device': {'name': '/dev/sda'}}}
+    ]
+    CephService._get_smart_data_by_device({'devid': '1', 'daemons': ['osd.1', 'mon.1']})
+    send_command.assert_has_calls([mock.call('mon', 'osd tree'),
+                                   mock.call('osd', 'smart', '1', devid='1')])
+
+    send_command.side_effect = [
+        {'nodes': [{'name': 'osd.1', 'status': 'down'}, {'name': 'mon.1', 'status': 'up'}]},
+        {'fake': {'device': {'name': '/dev/sda'}}}
+    ]
+    CephService._get_smart_data_by_device({'devid': '1', 'daemons': ['mon.1', 'osd.1']})
+    send_command.assert_has_calls([mock.call('mon', 'osd tree'),
+                                   mock.call('mon', 'device get-health-metrics', '1', devid='1')])
```

**Review by alfonsomthd — CHANGES_REQUESTED** [review]

File: src/pybind/mgr/dashboard/tests/test_ceph_service.py (hunk 1 above)
Comment (aaSharma14) [reply src/pybind/mgr/dashboard/tests/test_ceph_service.py]: Thanks a lot @alfonsomthd . Added this patch to the tests.

**Review by avanthakkar — APPROVED** [review]
LGTM!

**Review by alfonsomthd — APPROVED** [review]

**Review by nizamial09 — APPROVED** [review]

**Pull request merged**